ui/pages/confirmations/components/confirm/info/approve/spending-cap/spending-cap.tsx

```diff
@@ -92,21 +92,21 @@ export const SpendingCap = ({
 
   const { pending } = useApproveTokenSimulation(transactionMeta, decimals);
 
-  if (pending) {
-    return <Container isLoading />;
-  }
-
   return (
     <ConfirmInfoSection data-testid="confirmation__approve-spending-cap-section">
       <ConfirmInfoRow label={t('accountBalance')}>
         <ConfirmInfoRowText text={`${accountBalance} ${tokenSymbol || ''}`} />
       </ConfirmInfoRow>
 
-      <SpendingCapGroup
-        tokenSymbol={tokenSymbol || ''}
-        decimals={decimals || '0'}
-        setIsOpenEditSpendingCapModal={setIsOpenEditSpendingCapModal}
-      />
+      {pending ? (
+        <Container isLoading />
+      ) : (
+        <SpendingCapGroup
+          tokenSymbol={tokenSymbol || ''}
+          decimals={decimals || '0'}
+          setIsOpenEditSpendingCapModal={setIsOpenEditSpendingCapModal}
+        />
+      )}
```

Comment:
Only the value above need to wait for pending to go away and not the whole component.